pkg/pool-weighted/contracts/smart/IndexPool.sol

```diff
@@ -190,26 +162,31 @@ contract IndexPool is BaseWeightedPool, ReentrancyGuard {
         uint256 startTime,
         uint256 endTime,
         uint256[] memory startWeights,
-        uint256[] memory endWeights
+        uint256[] memory endWeights,
+        IERC20[] memory tokens
     ) internal virtual {
-        bytes32 newPoolState = _poolState;
-
         uint256 normalizedSum = 0;
+        bytes32 tokenState;
+
         for (uint256 i = 0; i < endWeights.length; i++) {
             uint256 endWeight = endWeights[i];
             _require(endWeight >= _MIN_WEIGHT, Errors.MIN_WEIGHT);
 
-            _tokenState[_tokens[i]] = _tokenState[_tokens[i]]
+            IERC20 token = tokens[i];
```

Comment:
Do we really need this variable? It is used only once...

Reply:
twice :P